c10/core/ScalarType.h

```diff
@@ -297,10 +297,40 @@ static inline ScalarType toUnderlying(ScalarType t) {
   }
 }
 
+static inline bool isSignedType(ScalarType t) {
+  if (t == ScalarType::Half) {
+    return true;
+  }
+  #define CASE_SIGNED(ctype, name) \
+    case ScalarType::name:                       \
+      return std::is_signed<ctype>();
+
+    switch (toUnderlying(t)) {
+      AT_FORALL_SCALAR_TYPES_WITH_COMPLEX_AND_QINTS(CASE_SIGNED)
```

Comment:
don't define this for QINTs, I'm not even sure what the correct answer is (just because the underlying representation is signed doesn't necessarily mean the value isn't signed.

You can define it for complex if you want, but you'll notice the existing code doesn't work either, so it's not a huge deal.